packages/content-management/table_list_view_table/src/components/user_filter_panel.tsx

```diff
@@ -0,0 +1,177 @@
+  defaultOptions: { queries: { retry: false, staleTime: 30 * 60 * 1000 } },
+});
+
+export const UserFilterContextProvider: FC<Context> = ({ children, ...props }) => {
+  if (!props.bulkGetUserProfiles) {
+    return <>{children}</>;
+  }
+
+  return (
+    <QueryClientProvider client={queryClient}>
+      <UserFilterContext.Provider value={props}>{children}</UserFilterContext.Provider>
+    </QueryClientProvider>
+  );
+};
+
+export const NULL_USER = 'no-user';
+
+export const UserFilterPanel: FC<{}> = () => {
+  const { euiTheme } = useEuiTheme();
+  const componentContext = React.useContext(UserFilterContext);
+  if (!componentContext)
+    throw new Error('UserFilterPanel must be used within a UserFilterContextProvider');
+  if (!componentContext.bulkGetUserProfiles)
+    throw new Error('UserFilterPanel must be used with a bulkGetUserProfiles function');
+
+  const { onSelectedUsersChange, selectedUsers } = componentContext;
+
+  const [isPopoverOpen, setPopoverOpen] = React.useState(false);
+  const [searchTerm, setSearchTerm] = React.useState('');
+
+  const query = useQuery({
+    queryKey: ['user-filter-suggestions', componentContext.allUsers],
+    queryFn: () => componentContext.bulkGetUserProfiles!(componentContext.allUsers),
+    enabled: isPopoverOpen,
+  });
+
+  const usersMap = React.useMemo(() => {
+    if (!query.data) return {};
+    return query.data.reduce((acc, user) => {
+      acc[user.uid] = user;
+      return acc;
+    }, {} as Record<string, UserProfile>);
+  }, [query.data]);
+
+  const visibleOptions = React.useMemo(() => {
+    if (!query.data || query.data.length === 0) return [];
+    // attach null to the end of the list to represent the "no creator" option
+    const users = [...query.data, null];
+
+    if (!searchTerm) {
+      return users;
+    }
+
+    return users.filter((user) => {
+      // keep the "no creator" option if it's selected
+      if (!user) {
+        return selectedUsers.includes(NULL_USER);
+      }
+
+      // keep the user if it's selected
+      if (selectedUsers.includes(user.uid)) return true;
+
+      // filter only users that match the search term
+      const searchString = (
+        user.uid +
+        user.user.username +
+        (user.user.email ?? '') +
+        (user.user.full_name ?? '')
+      ).toLowerCase();
+      return searchString.includes(searchTerm.toLowerCase());
+    });
+  }, [query.data, searchTerm, selectedUsers]);
+
+  const noUsersTip = (
+    <EuiIconTip
+      aria-label="Additional information"
+      position="bottom"
+      type="questionInCircle"
+      color="inherit"
+      iconProps={{ style: { verticalAlign: 'text-bottom', marginLeft: 2 } }}
+      css={{ textWrap: 'balance' }}
+      content={
+        <FormattedMessage
+          id="contentManagement.tableList.listing.userFilter.emptyMessageTooltip"
+          defaultMessage="Creators are assigned when dashboards are created, (for all dashboards created after version 8.14)."
```

Comment:
Not sure if the comma is supposed to be there?

```suggestion
          defaultMessage="Creators are assigned when dashboards are created (for all dashboards created after version 8.14)."
```

cc @amyjtechwriter